looker/resource_looker_role_groups.go

```diff
@@ -0,0 +1,220 @@
+	api := c.(*sdk.LookerSDK)
+
+	// read groups that are set on the role
+	roleID := d.Get("role_id").(string)
+	lookerGroupIDs, groupsErr := getGroupsOnRole(api, roleID)
+	if groupsErr != nil {
+		return diag.FromErr(groupsErr)
+	}
+
+	// read groups to be set by this resource
+	gIDs, ok := d.Get("group_ids").(*schema.Set)
+	if !ok {
+		return diag.Errorf("attribute group_ids is not of type *schema.Set")
+	}
+	groupIDs, gErr := conv.SchemaSetToSliceString(gIDs)
+	if gErr != nil {
+		return diag.FromErr(gErr)
+	}
+
+	diff := slice.Diff(lookerGroupIDs, groupIDs)
+
+	result := multierror.Append(
+		d.Set("role_id", roleID),
+		d.Set("group_ids", slice.Delete(lookerGroupIDs, diff)), // delete any group ids that are not in the terraform state
+	)
+
+	return diag.FromErr(result.ErrorOrNil())
+}
+
+func resourceRoleGroupsUpdate(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	o, n := d.GetChange("group_ids")
+
+	// get old and new groups set on this resource
+	old, ok := o.(*schema.Set)
+	if !ok {
+		return diag.Errorf("old role_id is not of type *schema.Set")
+	}
+	oldIDs, oErr := conv.SchemaSetToSliceString(old)
+	if oErr != nil {
+		return diag.FromErr(oErr)
+	}
+	new, ok := n.(*schema.Set)
+	if !ok {
+		return diag.Errorf("new role_id is not of type *schema.Set")
+	}
+	newIDs, nErr := conv.SchemaSetToSliceString(new)
+	if nErr != nil {
+		return diag.FromErr(nErr)
+	}
+
+	// read groups already set on this role
+	roleID := d.Get("role_id").(string)
+	lookerGroupIDs, groupsErr := getGroupsOnRole(api, roleID)
+	if groupsErr != nil {
+		return diag.FromErr(groupsErr)
+	}
+
+	// diff between what was has changed in the state and what is in looker
+	diff := slice.Diff(oldIDs, lookerGroupIDs)
+
+	// append
+	_, setErr := api.SetRoleGroups(roleID, append(diff, newIDs...), nil)
+	if setErr != nil {
+		return diag.FromErr(setErr)
+	}
+
+	d.SetId(strings.Join(append(newIDs, roleID), "_"))
+
+	return resourceRoleGroupsRead(ctx, d, c)
+}
+
+func resourceRoleGroupsDelete(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	// groups
+	gIDs, ok := d.Get("group_ids").(*schema.Set)
+	if !ok {
+		return diag.Errorf("attribute role_id is not of type *schema.Set")
+	}
+	groupIDs, rErr := conv.SchemaSetToSliceString(gIDs)
+	if rErr != nil {
+		return diag.FromErr(rErr)
+	}
+
+	roleID := d.Get("role_id").(string)
+	lookerGroupIDs, groupsErr := getGroupsOnRole(api, roleID)
+	if groupsErr != nil {
+		return diag.FromErr(groupsErr)
+	}
+
+	_, setErr := api.SetRoleGroups(roleID, slice.Diff(lookerGroupIDs, groupIDs), nil)
+	if setErr != nil {
+		return diag.FromErr(setErr)
+	}
+
+	return nil
+}
+
+func resourceRoleGroupImport(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	// id is delimited using `_`, eg. <role_id>_<group_ids>
+	u := regexp.MustCompile(`_`)
```

Comment:
Looking at existing importer functions, why are we splitting with `regexp` instead of `strings`?

Reply:
Funnily enough I was looking at the google provider and following their convention! Splitting using `strings` will do the job so I'll keep it simple and do that.